node/cmd/ccq/http.go

```diff
@@ -43,27 +46,37 @@ func (s *httpServer) handleQuery(w http.ResponseWriter, r *http.Request) {
 
 	// Set CORS headers for the preflight request
 	if r.Method == http.MethodOptions {
-
 		w.Header().Set("Access-Control-Allow-Methods", "PUT, POST")
 		w.Header().Set("Access-Control-Allow-Headers", "Content-Type, X-Api-Key")
 		w.Header().Set("Access-Control-Max-Age", "3600")
 		w.WriteHeader(http.StatusNoContent)
 		return
 	}
-	var q queryRequest
-	err := json.NewDecoder(r.Body).Decode(&q)
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusBadRequest)
-		return
-	}
 
 	// There should be one and only one API key in the header.
-	apiKey, exists := r.Header["X-Api-Key"]
-	if !exists || len(apiKey) != 1 {
-		s.logger.Debug("received a request without an api key", zap.Stringer("url", r.URL), zap.Error(err))
+	apiKeys, exists := r.Header["X-Api-Key"]
+	if !exists || len(apiKeys) != 1 {
+		s.logger.Debug("received a request with the wrong number of api keys", zap.Stringer("url", r.URL), zap.Int("numApiKeys", len(apiKeys)))
 		http.Error(w, "api key is missing", http.StatusUnauthorized)
 		return
 	}
+	apiKey := apiKeys[0]
+
+	// Make sure the user is authorized before we go any farther.
+	_, exists = s.permissions[strings.ToLower(apiKey)]
```

Comment:
we do this `ToLower` in two spots now (here and at the top of validate). perhaps just do that once in the line above?

Reply:
Will do this in the next PR.